xds/server.go

```diff
@@ -317,18 +322,66 @@ func (s *GRPCServer) GracefulStop() {
 	}
 }
 
+func routeAndProcess(ctx context.Context) error {
+	conn := transport.GetConnection(ctx)
+	cw, ok := conn.(*server.ConnWrapper)
+	if !ok {
+		return errors.New("missing connection wrapper in incoming context")
+	}
+	mn, ok := grpc.Method(ctx)
+	if !ok {
+		return errors.New("missing method name in incoming context")
+	}
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return errors.New("missing metadata in incoming context")
+	}
+	// Assuming to just take the first value defined.
+	authority := md.Get(":authority")
+	vh := xdsclient.FindBestMatchingVirtualHostServer(authority[0], cw.VirtualHosts)
+	if vh == nil {
+		return status.Error(codes.Unavailable, "the incoming RPC did not match a configured Virtual Host")
+	}
+
+	var rwi *xdsclient.RouteWithInterceptors
+	for _, r := range vh.Routes {
+		if r.M.Match(iresolver.RPCInfo{
```

Comment:
Given this rpcInfo is the same for every iteration of the for loop, you could initialize it before the loop. That will make this condition read much easier.

Reply:
Sure thing. Changed to initialize before loop once.